copy/compression.go

```diff
@@ -129,13 +129,13 @@ func (ic *imageCopier) bpcCompressUncompressed(stream *sourceStream, detected bp
 	if ic.c.dest.DesiredLayerCompression() == types.Compress && !detected.isCompressed {
 		logrus.Debugf("Compressing blob on the fly")
 		var uploadedAlgorithm *compressiontypes.Algorithm
-		if ic.c.compressionFormat != nil {
-			uploadedAlgorithm = ic.c.compressionFormat
+		if ic.compressionFormat != nil {
+			uploadedAlgorithm = ic.compressionFormat
 		} else {
 			uploadedAlgorithm = defaultCompressionFormat
 		}
 
-		reader, annotations := ic.c.compressedStream(stream.reader, *uploadedAlgorithm)
+		reader, annotations := ic.c.compressedStream(stream.reader, *uploadedAlgorithm, ic.compressionLevel)
```

Comment:
Please make `compressedStream` (and `compressGoroutine`) a method on `imageCopier` instead of adding a parameter.

(Historically, IIRC the code was written to use the more general `copier` when available, for possible generalization to non-`imageCopier` uses, but that doesn’t really reflect any coherent design thought; that generalization would now hinder us, so let’s specialize. Arguably many of these single-purpose utilities only used by `copySingleImage` could be methods of `imageCopier`, not `copier`.)

Reply:
Done.